activerecord/lib/active_record/railties/databases.rake

```diff
@@ -77,6 +77,8 @@ db_namespace = namespace :db do
   namespace :migrate do
     # desc  'Rollbacks the database one migration and re migrate up (options: STEP=x, VERSION=x).'
     task redo: [:environment, :load_config] do
+      raise "Empty VERSION provided" if !ENV["VERSION"].nil? && ENV["VERSION"].empty?
```

Comment:
Oh, but the behavior is now different. I think it should be `raise "Empty VERSION provided" if ENV["VERSION"] && ENV["VERSION"].empty?`